api/app_analytics/analytics_db_service.py

```diff
@@ -14,31 +14,103 @@
     FeatureEvaluationBucket,
     Resource,
 )
+from dateutil.relativedelta import relativedelta
 from django.conf import settings
 from django.db.models import Sum
 from django.utils import timezone
 
 from environments.models import Environment
 from features.models import Feature
+from organisations.models import Organisation
 
 ANALYTICS_READ_BUCKET_SIZE = 15
 
 
 def get_usage_data(
-    organisation, environment_id=None, project_id=None
-) -> List[UsageData]:
+    organisation: Organisation,
+    environment_id: int | None = None,
+    project_id: int | None = None,
+    period: str | None = None,
```

Comment:
How about we define a type like `Literal["current_billing_period", "previous_billing_period"]` for this annotation?